src/hotspot/share/prims/jvmtiExport.cpp

```diff
@@ -1107,8 +1107,8 @@ class JvmtiCompiledMethodLoadEventMark : public JvmtiMethodEventMark {
  public:
   JvmtiCompiledMethodLoadEventMark(JavaThread *thread, nmethod *nm, void* compile_info_ptr = NULL)
           : JvmtiMethodEventMark(thread,methodHandle(thread, nm->method())) {
-    _code_data = nm->insts_begin();
-    _code_size = nm->insts_size();
+    _code_data = nm->code_begin();
+    _code_size = (jint)pointer_delta(nm->code_end(), nm->code_begin(), sizeof(char));
```

Comment:
Could CodeBlob::code_size be used instead?:
```
  int code_size() const                          { return           code_end()       -           code_begin();       }
```

Reply:
It certainly could.  I was just following the pattern used by JvmtiExport::post_dynamic_code_generated but maybe it's better to use the real nmethod API.  I'll change it.